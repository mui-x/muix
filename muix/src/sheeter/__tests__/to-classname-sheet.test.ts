import { atomizeSheet } from "reactxx-sheeter";
import { initPlatform, ts, Shape } from "reactxx-tests";

export const createSheet = () =>
  (ts.sheet = {
    root: (ts.view = [
      {
        margin: 4,
        padding: 4,
        $web: (ts.web = {
          margin: 10
        })
      },
      {
        $native: (ts.nativeView = {
          padding: 10
        })
      }
    ]),
    label: (ts.text = {
      color: "blue"
    }),
    webOnly: {
      $web: (ts.web = {
        color: "red",
        padding: 20
      })
    },
    nativeOnly: {
      $native: (ts.nativeView = {
        margin: 20
      })
    }
  });

describe("TO CLASSNAMES, sheet", () => {
  it("NATIVE", () => {
    initPlatform(false);
    const sheet = atomizeSheet<Shape>(createSheet());
    `
  ##########################################
  ##########################################
  #
  #  NATIVE
  #
  ##########################################
  ##########################################
  `;
    expect(sheet).toMatchInlineSnapshot(`
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "color: blue /*label*/",
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "nativeOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "margin: 20 /*nativeOnly/$native*/",
        ],
      },
    ],
    "name": "nativeOnly",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "margin: 4 /*root[0]*/",
          "padding: 4 /*root[0]*/",
        ],
      },
      Object {
        "atomicArray": Array [
          "padding: 10 /*root[1]/$native*/",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`);
  });
  it("WEB", () => {
    initPlatform(true);
    const sheet = atomizeSheet<Shape>(createSheet());
    `
  ##########################################
  ##########################################
  #
  #  WEB
  #
  ##########################################
  ##########################################
  `;
    expect(sheet).toMatchInlineSnapshot(`
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".d { color:blue /*label*/ }",
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".a { margin:4px /*root[0]*/ }",
          ".b { padding:4px /*root[0]*/ }",
        ],
      },
      Object {
        "atomicArray": Array [
          ".c { margin:10px /*root[0]/$web*/ }",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
  "webOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".e { color:red /*webOnly/$web*/ }",
          ".f { padding:20px /*webOnly/$web*/ }",
        ],
      },
    ],
    "name": "webOnly",
    "~": "c",
  },
}
`);
  });
});
