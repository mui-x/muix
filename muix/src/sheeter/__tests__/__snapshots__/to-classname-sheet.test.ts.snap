// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TO CLASSNAMES, sheet NATIVE 1`] = `
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "color: blue /*sheet.label*/",
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "nativeOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "margin: 20 /*sheet.nativeOnly/$native*/",
        ],
      },
    ],
    "name": "nativeOnly",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "margin: 4 /*sheet.root[0]*/",
          "padding: 4 /*sheet.root[0]*/",
        ],
      },
      Object {
        "atomicArray": Array [
          "padding: 10 /*sheet.root[1]/$native*/",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`;

exports[`TO CLASSNAMES, sheet WEB 1`] = `
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".d { color:blue /*sheet.label*/ }",
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".a { margin:4px /*sheet.root[0]*/ }",
          ".b { padding:4px /*sheet.root[0]*/ }",
        ],
      },
      Object {
        "atomicArray": Array [
          ".c { margin:10px /*sheet.root[0]/$web*/ }",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
  "webOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".e { color:red /*sheet.webOnly/$web*/ }",
          ".f { padding:20px /*sheet.webOnly/$web*/ }",
        ],
      },
    ],
    "name": "webOnly",
    "~": "c",
  },
}
`;
